### ISSUE TITLE
Change of component type of AbstractTensor for symbolic calculations

### ISSUE THREAD

**Issue by jfbarthelemy** [Ferrite-FEM/Tensors.jl#153]
I have been trying to use the library in the framework of symbolic calculations. Unfortunately it doesn't work since the type of components of AbstractTensor is restricted to Real
```
abstract type AbstractTensor{order, dim, T <: Real} <: AbstractArray{T, order} end
```
However changing `T <: Real` into `T <: Number` seems to solve the problem insofar as it allows to use `Tensors.jl` together with `SymPy.jl`. Would it be relevant to make this change on the official repository?
Thank you very much.

**Comment by fredrikekre** [comment]
Is that all that needs to change? That's pretty cool in that case.

**Comment by KristofferC** [comment]
Sounds fine to me. I cannot recall why `<: Real` was chosen.

**Comment by jfbarthelemy** [comment]
Well it is the only change I had to make in the source in order to use it without any problem with symbolic values using `SymPy.jl`.

I had also to force the `offdiagscale` to `sympy.sqrt(2)` in `tomandel`  function so as to keep the symbolic value of `√2` in the Mandel view of fourth order tensors since `Sym(√2)` as current defaut value in the source (`T(√2)` with `T=Sym`) gives a decimal representation of `√2` before conversion to `Sym`. If I may, I would suggest in addition changing the default value
```
offdiagscale=T(√2)
```
into
```
offdiagscale=√(T(2))
```
I guess this would not change anything for numerical values and would allow to keep a nice symbolic `√2` by default when using `Sympy`.

I did not review the whole library and I'm rather new to Julia so I can't really ensure that these changes wouldn't have any consequence in some other applications.
Thanks a lot again for sharing this library.
